src/backend/src/lib.rs

```diff
@@ -511,9 +515,33 @@ fn list_custom_tokens() -> Vec<CustomToken> {
 
 #[update(guard = "caller_is_not_anonymous")]
 #[allow(clippy::needless_pass_by_value)]
-#[allow(unused_variables)]
-fn add_user_credential(request: AddUserCredentialRequest) {
-    // TODO: Implement https://dfinity.atlassian.net/browse/GIX-2649
+fn add_user_credential(request: AddUserCredentialRequest) -> Result<(), AddUserCredentialError> {
```

Comment:
For consistency, given that no other public functions returns a result, shouldn't this function trap as well in case of error?

Reply:
I wanted to give feedback to the frontend and show different UX. I believe that an error with the credentials should be reported to the user so that they can report it to the issuer. Versus a problem with Oisy backend can have a generic error to try again.

And I don't think that using trap to differentiate the errors is very developer friendly.

Reply:
Developer needs to try / catch anyway but works for me if the feature requires to display diffent UX in case of errors.